From 36e0db51ef1cba8ac21765fa8cb69d007d51b31a Mon Sep 17 00:00:00 2001
From: Lars Kellogg-Stedman <lars@redhat.com>
Date: Tue, 12 Jun 2018 14:35:52 -0400
Subject: [PATCH 2/4] [WIP] enable support for openidc federation in keystone

This enables the support for OpenIDC federation in
keystone::federation::openidc.

Change-Id: Id2ef3558a359883bf3182f50d6a082b1789a900a
(cherry picked from commit a90959a2bff27ba10e78403315237c67f8c47e2d)
---
 manifests/profile/base/keystone.pp | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/manifests/profile/base/keystone.pp b/manifests/profile/base/keystone.pp
index 628b032d..eb1acf80 100644
--- a/manifests/profile/base/keystone.pp
+++ b/manifests/profile/base/keystone.pp
@@ -142,6 +142,13 @@
 #   (Optional) Whether _member_ role is managed or not (required for Horizon).
 #   Defaults to hiera('keystone_enable_member', false)
 #
+# [*keystone_federation_enabled*]
+#   (Optional) Enable federated identity support
+#   Defaults to hiera('keystone_federation_enabled', false)
+#
+# [*keystone_openidc_enabled*]
+#   (Optional) Enable OpenIDC federation
+#   Defaults to hiera('keystone_openidc_enabled', false)
 class tripleo::profile::base::keystone (
   $admin_endpoint_network         = hiera('keystone_admin_api_network', undef),
   $bootstrap_node                 = hiera('bootstrap_nodeid', undef),
@@ -171,6 +178,8 @@ class tripleo::profile::base::keystone (
   $extra_notification_topics      = [],
   $step                           = Integer(hiera('step')),
   $keystone_enable_member         = hiera('keystone_enable_member', false),
+  $keystone_federation_enabled    = hiera('keystone_federation_enabled', false),
+  $keystone_openidc_enabled       = hiera('keystone_openidc_enabled', false),
 ) {
   if $::hostname == downcase($bootstrap_node) {
     $sync_db = true
@@ -256,6 +265,14 @@ class tripleo::profile::base::keystone (
         create_domain_entry => $manage_domain,
       })
     }
+
+    if $keystone_federation_enabled {
+      include ::keystone::federation
+    }
+
+    if $keystone_openidc_enabled {
+      include ::keystone::federation::openidc
+    }
   }
 
   if $step >= 4 and $manage_db_purge {
-- 
2.18.0.rc2

